apps/login/src/app/api/loginname/route.ts

```diff
@@ -104,14 +107,39 @@ export async function POST(request: NextRequest) {
           loginSettings?.allowRegister &&
           loginSettings?.allowUsernamePassword
         ) {
-          const params: any = { organization };
+          let orgToRegisterOn: string | undefined = organization;
+
+          if (
+            !orgToRegisterOn &&
+            loginName &&
+            ORG_SUFFIX_REGEX.test(loginName) &&
+            loginSettings.allowDomainDiscovery
```

Comment:
From where do we have this org? we query it later using the suffix.

BTW, I already had Spaghetti